src/mlpack/core/util/timers.cpp

```diff
@@ -69,15 +61,18 @@ bool Timers::GetState(std::string timerName)
 
 void Timers::PrintTimer(const std::string& timerName)
 {
-  timeval& t = timers[timerName];
-  Log::Info << t.tv_sec << "." << std::setw(6) << std::setfill('0')
-      << t.tv_usec << "s";
+  long long int totalDuration = timers[timerName].count();
+  // Converting microseconds to seconds
+  long long int totalDurationSec = totalDuration / 1e6;
+  long long int totalDurationMicroSec = totalDuration % 1000000;
```

Comment:
Why not use `std::chrono::microseconds::rep` here? That would avoid any casting issues.